command/kv_helpers.go

```diff
@@ -110,8 +110,24 @@ func addPrefixToVKVPath(p, mountPath, apiPrefix string) string {
 	case p == mountPath, p == strings.TrimSuffix(mountPath, "/"):
 		return path.Join(mountPath, apiPrefix)
 	default:
-		p = strings.TrimPrefix(p, mountPath)
-		return path.Join(mountPath, apiPrefix, p)
+		tp := strings.TrimPrefix(p, mountPath)
+		for {
+			// If the entire mountPath is included in the path, we are done
+			if tp != p {
+				break
+			}
+			// Trim the parts of the mountPath that are not included in the
+			// path, for example, in cases where the mountPath contains
+			// namespaces which are not included in the path.
+			partialMountPath := strings.SplitN(mountPath, "/", 2)
+			if len(partialMountPath) == 1 || partialMountPath[1] == ""{
```

Comment:
We probably should guard against the length prior to checking `partialMountPath[1]`. This could likely experience an index out of bounds error.

Reply:
In the case that the length of `partialMountPath` is 0, the check for `partialMountPath[1] == ""` will still be performed and thus result in an index out of bounds error. Here is an [example in action](https://play.golang.org/p/-zt_ocPgDpA).

Reply:
Ah, It should be lessEqual than 1. Thank you!